### DISCUSSION TITLE
Posting to Conductor Queue from an outside application

### DISCUSSION THREAD

**Discussion by krishnapyde** [Netflix/conductor#2803] [Q&A]
Currently, an event can be produced to a Conductor Queue via an EVENT task. Once done, using an Event Handler, we can either start a workflow or change the status of a task in a workflow.

My question is, can we produce an Event to the Conductor Queue from another application? 
Let's say, I have a Customer facing REST endpoint, and whenever that endpoint is invoked, I register that call and then want to start a workflow in the conductor. Instead of invoking this conductor workflow via API/HTTP call, I want to produce an event from my code. Is this even possible? if yes, can you point me to any documentation around this?

**Comment by v1r3n** [comment]
Currently this is not exposed as a service, however it should be possible to expose an API endpoint in Conductor that allows publishing event to internal Conductor queues.

If your use cases is to trigger workflow asynchronously, how about an async http endpoint that can be used to trigger a workflow from an external service? 

Would love to discuss more.

**Comment by aravindanr** [comment]
fwiw, Conductor supports SQS and AMQP as its event source. To start a workflow, a message can be posted in SQS and an event handler can be configured in Conductor to start workflow based on the data in the SQS message. Let us know if you need help configuring SQS.

**Reply by krishnapyde** [reply]
Unfortunately, Using SQS is not an option.
but regarding AMQP, can any implementation of AMQP be used (Apache ActiveMQ / JMS)

**Reply by apanicker-nflx** [reply]
@ritu-p @venkathangella Can you please provide some information about AMQP implementation usage?

**Reply by NoahLerner** [reply]
Hey @apanicker-nflx, what should expected throughput per host be for starting workflows using AMQP? We're getting very slow event handling on our queues, something like 3-5 messages per second per consumer. We're running simple workflows for performance testing which contain only a single Lambda function that just does `return;`.

Either increasing throughput per consumer, or somehow implementing more consumers per host are both acceptable options but we're at a loss on how to do this. Thanks.

**Reply by apanicker-nflx** [reply]
You could increase the number of threads consuming events on your host. The property that controls this is `conductor.app.eventProcessorThreadCount`. This defaults to 2.  

For reference, see [here](https://github.com/Netflix/conductor/blob/main/core/src/main/java/com/netflix/conductor/core/events/DefaultEventProcessor.java#L91). To get higher throughput on event processors, I would recommend deploying the event processors on a separate cluster and horizontally scaling out this cluster.
